### PR TITLE
nvuApp: registration page

### PR DESCRIPTION
**Pull request by denisdimitrov**
Registration page with test and everything, I will open new PR for the login page because the code will become too massive. Apologies for the large changes, most of them are just from formatting. Here are the files I worked on if it makes it any easier:
TutorRepository,
TutorHandler,
PersistentTutorRepository,
JettyMain,
and the html files.

### CODE REVIEW

File: src/main/java/JettyMain.java [hunk 1]
```diff
@@ -23,58 +24,59 @@
  * @author Vasil Mitov <v.mitov.clouway@gmail.com>
  */
 public class JettyMain {
-  public static void main(String[] args) {
+    public static void main(String[] args) {
```
Comment (mgenov) [inline src/main/java/JettyMain.java]: The tab should be 2 spaces not 4.


Company's code-style is available at: https://github.com/clouway/teamspace/blob/master/clouway_codestyle.xml

File: src/main/resources/http/servlet/register.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            return false;
+        }
+        return true;
+    }
+
+    function password_validate(password, x, y) {
+        var letters = /[0-9a-zA-Z]+$/;
+        var passlength = password.value.length;
+        if (passlength > y || passlength < x) {
+            alert("Дължината на паролата трябва да бъде между " + x + " и " + y + " символа.");
```
Comment (mgenov) [inline src/main/resources/http/servlet/register.js]: It's not user friendly messages to be shown as alerts.

File: src/main/java/JettyMain.java (hunk 1 above)
Comment (denisdimitrov) [inline src/main/java/JettyMain.java]: Just downloaded it, I will use it from now on.

File: src/main/resources/http/servlet/register.js (hunk 2 above)
Comment (denisdimitrov) [inline src/main/resources/http/servlet/register.js]: Is console.log a better alternative?

**Comment by denisdimitrov** [comment]
All changes applied. Clouway codestyle in use, the invalid messages are no longer alerts and all commits are squashed.

File: src/main/java/persistent/adapter/ConnectionProvider.java [hunk 3]
```diff
@@ -11,17 +11,19 @@
...
+    this.password = password;
   }
 
   @Override
   public Connection get() {
     Connection connection = null;
     try {
       Class.forName("com.mysql.jdbc.Driver").newInstance();
-      connection = DriverManager.getConnection("jdbc:mysql://localhost/" + database + "?user=root&password=clouway.com");
+      connection = DriverManager.getConnection("jdbc:mysql://localhost/" + database + "?user=root&password=" + password + "");
```
Comment (mgenov) [inline src/main/java/persistent/adapter/ConnectionProvider.java]: In #21 this change was already made and host was extracted as parameter too.

File: src/test/java/core/ResponseReader.java [hunk 4]
```diff
@@ -7,18 +7,18 @@
...
-        String page = "";
-        try (ByteArrayOutputStream out = new ByteArrayOutputStream()) {
-            ByteStreams.copy(response.body(), out);
-            page = out.toString();
-        } catch (IOException e) {
-            e.printStackTrace();
-        }
-        return page;
+  public String read(Response response) {
+    String page = "";
```
Comment (mgenov) [inline src/test/java/core/ResponseReader.java]: Could be simplified with: 

```java
return new String(ByteStreams.toByteArray(response.body()), StandardCharsets.UTF_8);
```

File: src/test/java/http/servlet/PageHandlerServletTest.java [hunk 5]
```diff
@@ -14,67 +13,64 @@
...
-      oneOf(resp).sendRedirect("/someOtherPage");
-    }});
-    pageHandlerServlet.doGet(req, resp);
-  }
+    @Rule
+    public JUnitRuleMockery context = new JUnitRuleMockery();
+
+
+    @Test
+    public void happyPath() throws Exception {
```
Comment (mgenov) [inline src/test/java/http/servlet/PageHandlerServletTest.java]: Still old formatting.

File: src/test/java/persistent/PersistentTutorRepositoryTest.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    String dropTable = "DROP TABLE TUTORS";
+    dataStore.update(dropTable);
+  }
+
+  @Test
+  public void happyPath() throws Exception {
```
Comment (mgenov) [inline src/test/java/persistent/PersistentTutorRepositoryTest.java]: You have missing scenarios here. Usually you have to test for: 0, 1 and many.

try to find when no tutors are registered
try to find tutor that is registered
try to find one tutor, when different one is registered

File: src/test/java/persistent/PersistentTutorRepositoryTest.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+/**
+ * @author Denis Dimitrov <denis.k.dimitrov@gmail.com>.
+ */
+public class PersistentTutorRepositoryTest {
+  private ConnectionProvider connectionProvider = new ConnectionProvider("nvuApp", "root");
+  private DataStore dataStore = new DataStore(connectionProvider);
+  private PersistentTutorRepository tutors = new PersistentTutorRepository(dataStore);
+
+  @Before
+  public void setUp() throws Exception {
```
Comment (mgenov) [inline src/test/java/persistent/PersistentTutorRepositoryTest.java]: Doesn't look like a good solution for the whole project, but as it's a first test maybe it's a good as start.

**Comment by denisdimitrov** [comment]
All of the required changes are applied.

**Comment by mgenov** [comment]
I don't see answers on some of my comments.

File: src/test/java/persistent/PersistentTutorRepositoryTest.java (hunk 6 above)
Comment (denisdimitrov) [inline src/test/java/persistent/PersistentTutorRepositoryTest.java]: Scenarios have been added.

File: src/main/java/persistent/adapter/ConnectionProvider.java (hunk 3 above)
Comment (denisdimitrov) [inline src/main/java/persistent/adapter/ConnectionProvider.java]: I will change my constructor and push the changes in the next 2-3 minutes.

File: src/test/java/persistent/PersistentTutorRepositoryTest.java (hunk 7 above)
Comment (denisdimitrov) [inline src/test/java/persistent/PersistentTutorRepositoryTest.java]: Should I change it or just keep it like this?

**Comment by denisdimitrov** [comment]
Changed the ConnectionProvider constructor and all of requested changes should be applied now.

File: src/test/java/persistent/PersistentTutorRepositoryTest.java (hunk 7 above)
Comment (mgenov) [inline src/test/java/persistent/PersistentTutorRepositoryTest.java]: I would like to see how it will go.

**Pull request merged**